templates/_helpers.tpl

```diff
@@ -343,9 +343,14 @@ Merge packs and virtualenvs from st2 with those from st2packs images
   command:
     - 'sh'
     - '-ec'
-    - |
-      /bin/cp -aR /opt/stackstorm/packs/. /opt/stackstorm/packs-shared &&
-      /bin/cp -aR /opt/stackstorm/virtualenvs/. /opt/stackstorm/virtualenvs-shared
+    - >
+      if command rsync; then
+        rsync -a /opt/stackstorm/packs/. /opt/stackstorm/packs-shared &&
+        rsync -a /opt/stackstorm/virtualenvs/. /opt/stackstorm/virtualenvs-shared;
+      else
+        /bin/cp -aR /opt/stackstorm/packs/. /opt/stackstorm/packs-shared &&
```

Comment:
`-a` is equivalent to `-dR --preserve=all`:
- `-d`: no dereference (preserve links)
- `-R`: copy directories recursively
- `--preserve=all`: preserve file attributes
  - `mode`
  - `ownership` for user and group
  - `timestamps`
  - `links` for hard links
  - `context` for security context
  - `xattr` for extended attributes

Of these we need `-dR --preserve=mode` (we need mode to maintain the execute bits of action files). Depending on which user is copying the file, we might also need `--preserve=ownership`.

Reply:
I think I'll push a commit that switches to `-RP --preserve=mode,timestamps,links,xattr` based on https://github.com/StackStorm/stackstorm-k8s/compare/rsync...Stealthii:stackstorm-k8s:rsync